### PR TITLE
Using mmap64 instead of plain mmap.

### PR DESCRIPTION
**Pull request by lvella**
~May actually fix~ Actually fixes issue #85, as per my test the first triggered it.

### CODE REVIEW

**Comment by adamreichold** [comment]
> May actually fix issue https://github.com/RazrFalcon/memmap2-rs/issues/85.

Could you verify that this fixes the issue for at least those platforms you care about?

**Comment by lvella** [comment]
Well, it seems macos doesn't have `mmap64`.

File: src/unix.rs [hunk 1]
```diff
@@ -50,13 +50,13 @@ impl MmapInner {
         let (map_len, map_offset) = Self::adjust_mmap_params(len as usize, alignment as usize)?;
 
         unsafe {
-            let ptr = libc::mmap(
+            let ptr = libc::mmap64(
```
Comment (adamreichold) [inline src/unix.rs:53]: We cannot use this unconditionally.

I suggest doing something like

```
#[cfg(all(target_os = "linux", not(target_env = "musl")))]
use libc::{mmap64 as mmap, off64_t as off_t};
```

and of course the opposite condition importing `mmap` and `off` directly.

Comment (adamreichold) [reply src/unix.rs:53]: Alternatively we could reproduce the full cfg attribute from [here](https://docs.rs/libc/latest/src/libc/unix/mod.rs.html#1557).

**Comment by adamreichold** [comment]
And indeed we should test Musl as part of the CI.

File: src/unix.rs (hunk 1 above)
Comment (lvella) [reply src/unix.rs:53]: What about this way?
```
#[cfg(target_env = "glibc")]
use libc::{mmap64 as mmap, off64_t as off_t};
```
Since this `mmap64` is a glibc thing, we theoretically have it covered for everything glibc.

Comment (lvella) [reply src/unix.rs]: It seems this is an Android thing, too.

Comment (adamreichold) [reply src/unix.rs]: It is a bit irritating, but I guess we should just reproduce those cfg attributes from libc here and use mmap64 whenever it is available.

Comment (lvella) [reply src/unix.rs]: It is a maze there. On Linux, it seems you were right: every lib but musl has `mmap64`.

Comment (adamreichold) [reply src/unix.rs]: My understanding of the libc source would have been

```rust
#[cfg(all(
    any(target_os = "linux", target_os = "l4re", target_os = "android", target_os = "emscripten"),
    not(target_env = "musl"),
))]
```

But as you point out, the libc source is somewhat complex and I might have misunderstood it.

At the end of the day we also test none of l4re, android or emscription. So I see two options, copy libc verbatim or limit ourselves to those combinations which we actually test.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -76,7 +78,7 @@ jobs:
         toolchain: stable-${{ matrix.target }}
```
Comment (adamreichold) [inline .github/workflows/main.yml:78]: I suspect this works better if we use `dtolnay/rust-toolchain@stable` and `with: targets: ${{ matrix.target }}` together with `cargo test --target=${{ matrix.target }}`.

Comment (lvella) [reply .github/workflows/main.yml]: Ok, now tests are passing, I tested it on my fork.

**Comment by RazrFalcon** [comment]
All good. Thanks. Will publish soon.
